index.js

```diff
@@ -106,10 +106,15 @@ module.exports = url => {
             app.methodToBeUsed(
                 url,
                 (req, res) => {
-                    res.send(handler({
+                    const response = handler({
                         url: req.url,
-                        query: req.query
-                    }));
+                        req: req,
+                        res: res
+                    });
+
+                    if (response) {
```

Comment:
I don't quite understand the motivation behind this check. How about ignore the response when not required rather than make this response optional?

Reply:
There will be times when it will not be a direct returned value that you need to respond to the client, for example when its an async process or when you want to return status codes other than 200.

So this check will see if anything is to be returned, else the called handler function will be responsible to respond.

It may have broken other parts of the code, we will probably need to find a better solution.